cloud-azure/src/main/java/com/sequenceiq/cloudbreak/cloud/azure/AzureInstanceConnector.java

```diff
@@ -68,22 +68,47 @@ public String getConsoleOutput(AuthenticatedContext authenticatedContext, CloudI
-                        statuses.add(new CloudVmInstanceStatus(vm, InstanceStatus.FAILED, throwable.getMessage()));
-                    })
-                    .doOnCompleted(() -> statuses.add(new CloudVmInstanceStatus(vm, InstanceStatus.STARTED)))
-                    .subscribeOn(Schedulers.io()));
+        List<Completable> completables = new ArrayList<>();
+        List<CloudVmInstanceStatus> currentStatuses = check(ac, vms);
+        for (CloudVmInstanceStatus vm : currentStatuses) {
+            if (vm.getStatus() == InstanceStatus.STARTED) {
+                doReboot(completables, ac, vm, statuses);
+            } else if (vm.getStatus() == InstanceStatus.STOPPED) {
+                doStart(completables, ac, vm, statuses);
+            } else {
+                LOGGER.error(String.format("Unable to reboot instance %s because of invalid status %s.",
+                        vm.getCloudInstance().getInstanceId(), vm.getStatus().toString()));
+            }
         }
-        Completable.merge(rebootCompletables).await();
+        Completable.merge(completables).await();
         return statuses;
     }
 
+    private void doReboot(List<Completable> completables, AuthenticatedContext ac, CloudVmInstanceStatus vm, List<CloudVmInstanceStatus> statuses) {
+        AzureClient azureClient = ac.getParameter(AzureClient.class);
+        String resourceGroupName = azureUtils.getResourceGroupName(ac.getCloudContext(), vm.getCloudInstance());
+        completables.add(azureClient.rebootVirtualMachineAsync(resourceGroupName, vm.getCloudInstance().getInstanceId())
+                .doOnError(throwable -> {
+                    LOGGER.error("Error happend on azure instance reboot: {}", vm, throwable);
+                    statuses.add(new CloudVmInstanceStatus(vm.getCloudInstance(), InstanceStatus.FAILED, throwable.getMessage()));
+                })
+                .doOnCompleted(() -> statuses.add(new CloudVmInstanceStatus(vm.getCloudInstance(), InstanceStatus.STARTED)))
+                .subscribeOn(Schedulers.io()));
+    }
+
+    private void doStart(List<Completable> completables, AuthenticatedContext ac, CloudVmInstanceStatus vm, List<CloudVmInstanceStatus> statuses) {
+        AzureClient azureClient = ac.getParameter(AzureClient.class);
+        String resourceGroupName = azureUtils.getResourceGroupName(ac.getCloudContext(), vm.getCloudInstance());
+        completables.add(azureClient.startVirtualMachineAsync(resourceGroupName, vm.getCloudInstance().getInstanceId())
+                .doOnError(throwable -> {
+                    LOGGER.error("Error happend on azure instance start: {}", vm, throwable);
+                    statuses.add(new CloudVmInstanceStatus(vm.getCloudInstance(), InstanceStatus.FAILED, throwable.getMessage()));
+                })
+                .doOnCompleted(() -> statuses.add(new CloudVmInstanceStatus(vm.getCloudInstance(), InstanceStatus.STARTED)))
+                .subscribeOn(Schedulers.io()));
+
+    }
```

Comment:
I think there is only one line difference in these 2 methods. caould you refactor it? I would do something like:
```
if (vm.getStatus() == InstanceStatus.STARTED) {
    doReboot(completables, vm, statuses, () -> azureClient.rebootVirtualMachineAsync(resourceGroupName, vm.getCloudInstance().getInstanceId()));
} else if (vm.getStatus() == InstanceStatus.STOPPED) {
    doReboot(completables, vm, statuses, () -> azureClient.startVirtualMachineAsync(resourceGroupName, vm.getCloudInstance().getInstanceId()));
}
```
and
```
private void doReboot(List<Completable> completables, CloudVmInstanceStatus vm, List<CloudVmInstanceStatus> statuses, Supplier<Completable> supplier) {
        completables.add(supplier.get()
                .doOnError(throwable -> {
                    LOGGER.error("Error happend on azure instance reboot: {}", vm, throwable);
                    statuses.add(new CloudVmInstanceStatus(vm.getCloudInstance(), InstanceStatus.FAILED, throwable.getMessage()));
                })
                .doOnCompleted(() -> statuses.add(new CloudVmInstanceStatus(vm.getCloudInstance(), InstanceStatus.STARTED)))
                .subscribeOn(Schedulers.io()));
    }
```